### PR TITLE
feat: support glob for overrides

### PR DESCRIPTION
**Pull request by SettingDust**

### CODE REVIEW

File: src/commonMain/kotlin/teksturepako/pakku/io/Glob.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.Path
+import kotlin.io.path.PathWalkOption
+import kotlin.io.path.walk
+
+@OptIn(ExperimentalPathApi::class)
+fun Path.listDirectoryEntriesRecursive(glob: String): List<Path>
+{
+    val matcher = FileSystems.getDefault().getPathMatcher("glob:${glob.removePrefix("./")}")
+    return walk(PathWalkOption.FOLLOW_LINKS, PathWalkOption.INCLUDE_DIRECTORIES).filter { matcher.matches(it) }.toList()
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/io/Glob.kt:14]: It probably shouldn't follow symlinks. Is there a reason to follow them?

File: src/commonMain/kotlin/teksturepako/pakku/api/actions/export/Export.kt [hunk 2]
```diff
@@ -270,11 +271,11 @@ suspend fun List<ExportRule>.produceRuleResults(
...
         acc + lockFile.getAllProjects().map {
             rule to RuleContext.ExportingProject(it, lockFile, configFile, workingSubDir)
-        } + configFile.getAllOverrides().map {
+        } + filterOverrides(configFile.getAllOverrides().expandWithGlob()).map {
             rule to RuleContext.ExportingOverride(it, OverrideType.OVERRIDE, lockFile, configFile, workingSubDir)
-        } + configFile.getAllServerOverrides().map {
+        } + filterOverrides(configFile.getAllServerOverrides().expandWithGlob()).map {
             rule to RuleContext.ExportingOverride(it, OverrideType.SERVER_OVERRIDE, lockFile, configFile, workingSubDir)
-        } + configFile.getAllClientOverrides().map {
+        } + filterOverrides(configFile.getAllClientOverrides().expandWithGlob()).map {
```
Comment (juraj-hrivnak) [inline src/commonMain/kotlin/teksturepako/pakku/api/actions/export/Export.kt:278]: Can you do this in the config file?

**Review by juraj-hrivnak — CHANGES_REQUESTED** [review]

File: src/commonMain/kotlin/teksturepako/pakku/io/Glob.kt (hunk 1 above)
Comment (SettingDust) [reply src/commonMain/kotlin/teksturepako/pakku/io/Glob.kt:14]: No reason to follow or not.

Comment (juraj-hrivnak) [reply src/commonMain/kotlin/teksturepako/pakku/io/Glob.kt:14]: Yes, no reason to follow.

**Pull request merged**